Source/NonlinearSolvers/NonlinearSolverLibrary.H

```diff
@@ -0,0 +1,15 @@
+#ifndef NONLINEAR_SOLVER_LIBRARY_H_
+#define NONLINEAR_SOLVER_LIBRARY_H_
+
+#include "PicardSolver.H"
+#include "NewtonSolver.H"
```

Comment:
Pls see above:
```suggestion
#include "PicardSolver.H"  // IWYU pragma: export
#include "NewtonSolver.H"  // IWYU pragma: export
```

Reply:
The `..Solver/...SolverLibrary.H` name appears a bit verbose.
Maybe we can name these two facade headers something like `ImplicitSolvers/Library.H` and `NonlinearSolver/Library.H`?

Reply:
Related to the other comment in `WarpX.H`:
We could also add a file for [forward declarations](https://warpx.readthedocs.io/en/latest/developers/repo_organization.html#c-includes) `_fwd.H` to speed up the common use of `WarpX.H`.